src/components/history-page/history-toolbar/history-view-mode-toggle-button.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useCallback } from 'react'
+import { ToggleButton, ToggleButtonGroup } from 'react-bootstrap'
+import { cypressId } from '../../../utils/cypress-attribute'
+import { ViewStateEnum } from './history-toolbar'
+import { useTranslation } from 'react-i18next'
+import { ForkAwesomeIcon } from '../../common/fork-awesome/fork-awesome-icon'
+import { useHistoryToolbarState } from './toolbar-context/use-history-toolbar-state'
+
+/**
+ * Toggles the view mode of the history entries between list and card view.
+ */
+export const HistoryViewModeToggleButton: React.FC = () => {
+  const { t } = useTranslation()
+  const [historyToolbarState, setHistoryToolbarState] = useHistoryToolbarState()
+
+  const onViewStateChange = useCallback(
+    (newViewState: ViewStateEnum) => {
+      setHistoryToolbarState((state) => ({
+        ...state,
+        viewState: newViewState
+      }))
+    },
+    [setHistoryToolbarState]
+  )
+
+  return (
+    <ToggleButtonGroup
+      type='radio'
+      name='options'
+      dir='ltr'
+      value={historyToolbarState.viewState}
+      className={'button-height'}
+      onChange={onViewStateChange}>
+      <ToggleButton className={'btn-light'} value={ViewStateEnum.CARD} title={t('landing.history.toolbar.cards')}>
```

Comment:
This seems to be missing, which is probably an indicator that the test are not complete here either
```suggestion
      <ToggleButton
        {...cypressId('history-mode-card')}
        className={'btn-light'}
        value={ViewStateEnum.CARD}
        title={t('landing.history.toolbar.cards')}>
```

Reply:
What seems to be missing?

Reply:
The `cypressId` call

Reply:
I won't add it because it wasn't there in the original code. Create an issue.

Reply:
https://github.com/hedgedoc/react-client/issues/1727